python/tank/platform/framework.py

```diff
@@ -215,28 +215,27 @@ def log_exception(self, msg):
 # Helper methods for loading frameworks
 #
 
+
 def setup_frameworks(engine_obj, parent_obj, env, parent_descriptor):
     """
     Checks if any frameworks are needed for the current item
     and in that case loads them - recursively
     """
-    
+
     # look into the environment, get descriptors for all frameworks that our item needs:
     framework_instance_names = validation.validate_and_return_frameworks(parent_descriptor, env)
-    
+
     # looks like all of the frameworks are valid! Load them one by one
-    for fw_inst_name in framework_instance_names:
-        
+    for fw_name, fw_inst_name in framework_instance_names:
+
         engine_obj.log_debug("%s - loading framework %s" % (parent_obj, fw_inst_name))
```

Comment:
I suggest adding some debug logging entry to highlight which name key is used when the framework gets added to the `bundle.frameworks` dict